.travis.yml

```diff
@@ -172,41 +172,41 @@ script:
-  - python -m pytest -v --durations=10 test/microbenchmarks.py
-  - python -m pytest -v --durations=10 test/stress_tests.py
-  - python -m pytest -v --durations=10 test/component_failures_test.py
-  - python -m pytest -v --durations=10 test/multi_node_test.py
-  - python -m pytest -v --durations=10 test/multi_node_test_2.py
-  - python -m pytest -v --durations=10 test/recursion_test.py
-  - python -m pytest -v --durations=10 test/monitor_test.py
-  - python -m pytest -v --durations=10 test/cython_test.py
-  - python -m pytest -v --durations=10 test/credis_test.py
-  - python -m pytest -v --durations=10 test/node_manager_test.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_global_state.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_queue.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_ray_init.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_mini.py
+
+  - python -m pytest -v --durations=10 python/ray/tests/test_basic.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_array.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_actor.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_autoscaler.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_tensorflow.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_failure.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_microbenchmarks.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_stress.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_component_failures.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_multi_node.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_multi_node_2.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_recursion.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_monitor.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_cython.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_credis.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_node_manager.py
   # TODO(yuhguo): object_manager_test.py requires a lot of CPU/memory, and
   # better be put in Jenkins. However, it fails frequently in Jenkins, but
   # works well in Travis. We should consider moving it back to Jenkins once
   # we figure out the reason.
-  - python -m pytest -v --durations=10 test/object_manager_test.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_object_manager.py
 
   # ray temp file tests
-  - python -m pytest -v --durations=10 test/tempfile_test.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_tempfile.py
 
   # ray debug tools tests
-  - python -m pytest -v --durations=10 test/debug_tools_test.py
+  - python -m pytest -v --durations=10 python/ray/tests/test_debug_tools.py
 
   # modin test files
-  - python python/ray/test/test_modin.py
+  - python python/ray/tests/test_modin.py
```

Comment:
I think now it's a good chance to follow the test discovery convention (see https://docs.pytest.org/en/latest/goodpractices.html#conventions-for-python-test-discovery) and change these file-based commands to to one single directory-based command (or a few sub-directories if needed).
The benefit is that we can only have one statement in this travis ci file, e.g., `pytest -v --duration=10 python/ray/tests`. And when someone adds a new test file, they don't need to modify here.

Reply:
Thanks! Doing so also revealed that the `test_queue.py` lacked a `teardown_module`, which I added in #4012. 

I notice that there was also a comment about `object_manager_test.py` and I left it as a separate command but I am open to suggestions